endpoints/openrtb2/video_auction.go

```diff
@@ -198,6 +203,9 @@ func (deps *endpointDeps) VideoAuctionEndpoint(w http.ResponseWriter, r *http.Re
 			labels.CookieFlag = pbsmetrics.CookieFlagYes
 		}
 	}
+	if labels.PubID == "UNKNOWN" && bidReq.Site != nil && bidReq.Site.Publisher != nil && bidReq.Site.Publisher.ID != "" {
```

Comment:
PubID can have three values at this point:  (1) "" from line 77, (2) bidReq.App.Publisher.ID from line 194, or (3) "UNKNOWN" from line 196.  The logic from this line suggests we want to set PubID to bidReq.Site.Publisher.ID only if bidReq.App was not nil but there was no bidReq.App.Publisher.ID.  If so, this if-statement should be modified and then used to replace line 196. If not, the logic here is incorrect.

Reply:
Yep, I knew there was a reason for me to have my `labels.PubID == "" &&` check here. I don't know when did I decide otherwise. Corrected